### DISCUSSION TITLE
Why `checkpoints.restore_checkpoint` does not raise an error?

### DISCUSSION THREAD

**Discussion by PgLoLo** [google/flax#1612] [Q&A]
From documentation of `flax.training.checkpoints.restore_checkpoint`:
>Returns:
>   Restored `target` updated from checkpoint file, or if no step specified and
>   no checkpoint files present, returns the passed-in `target` unchanged.
>   If a file path is specified and is not found, the passed-in `target` will be
>   returned. This is to match the behavior of the case where a directory path
>   is specified but the directory has not yet been created.

Why silently hiding the checkpoint absence (which is a great potential for a bug: by path misspecification, step misspecification, etc.) is a good idea? Makes no sense to me, is there any logic behind this decision?

**Comment by jheek** [comment]
Usually a directory is specified in which to save checkpoints and restore in case of pre-emption of the train script.
But I agree that if a specific step/path is specified then restore_checkpoint should raise an error.
Quickly scanning the code it seems like an error is raised when a step is provided explicitly but not if a checkpoint file is passed explicitly (just a logging info). I agree that looks like a silent foot-gun to me.

**Reply by jheek** [reply]
Created issue #1631
